### DISCUSSION TITLE
suggestion: python examples should be timezone aware

### DISCUSSION THREAD

**Discussion by mgalbright** [open-meteo/open-meteo#650] [Ideas]
I suggest the python api examples be modified (slightly) to make them timezone aware. In my opinion, it would improve clarity.

For example, when I visit the open-meteo docs page and specify a time zone like America/Chicago, as in this [URL](https://open-meteo.com/en/docs/#latitude=41.85&longitude=-87.65&timezone=America%2FChicago), and then click the tab listing Python code that calls the API, the generated code looks like
```python
hourly_data = {"date": pd.date_range(
	start = pd.to_datetime(hourly.Time(), unit = "s"),
	end = pd.to_datetime(hourly.TimeEnd(), unit = "s"),
	freq = pd.Timedelta(seconds = hourly.Interval()),
	inclusive = "left"
)}
```

When I run the full code to retrieve an hourly forecast, the first datetime returned is "2024-02-11 06:00:00".  This datetime is not time-zone aware.  Furthermore, I think that creates ambiguity/potential confusion for the user, since the user specified the America/Chicago time zone, but this first date time (hour 00 in the Chicago time zone) is expressed in the GMT timezone with hour 06.  That may be the desired behavior, but it creates the potential for confusion to the user.  (E.g., they could assume, incorrectly, that the date time is in the Chicago time zone, which they requested.)

I suggest modifying the code (both for hourly and daily data) to make the datatimes time-zone aware, like this:
```python
hourly_data = {"date_utc": pd.date_range(
	start = pd.to_datetime(hourly.Time(), unit = "s", utc=True),
	end = pd.to_datetime(hourly.TimeEnd(), unit = "s", utc=True),
	freq = pd.Timedelta(seconds = hourly.Interval()),
	inclusive = "left"
)}
```
Optionally, the example code could also compute the local time, like
```python
hourly_data['date_local'] = hourly_data['date_utc'].dt.tz_convert(response.Timezone().decode())
```

In my opinion that makes python results much clearer.  E.g. for the first forecast result, the date_utc value "2024-02-11 06:00:00+00:00" is clearly in the GMT time zone, and date_local has a value "2024-02-11 00:00:00-06:00" which is also unambiguous -- it's hour 00 in the America/Chicago time zone.  

This change could be made to both the hourly and daily data values for the python example code on the [open-meteo docs](https://open-meteo.com/en/docs) page, as well as the examples for the [open-meteo python_requests](https://github.com/open-meteo/python-requests) page.

**Comment by patrick-zippenfenig** [comment]
Thanks! I added `utc=True` in https://github.com/open-meteo/open-meteo-website/commit/b9988970040dcdbcd3995e3d7650038f07c55ada. 

For date_local I am considering to add the utc_offset_seconds attribute:
```python
hourly_data = {"date_local": pd.date_range(
	start = pd.to_datetime(hourly.Time() + response.UtcOffsetSeconds(), unit = "s", utc = True),
	end = pd.to_datetime(hourly.TimeEnd() + response.UtcOffsetSeconds(), unit = "s", utc = True),
	freq = pd.Timedelta(seconds = hourly.Interval()),
	inclusive = "left"
)}
```

This abuses timezone calculations, but it is certain that the same UTC offset was used as on server side. Usually, I do not trust timezone calculations at two places at the same time and instead use attributes like `utc_offset_seconds`. Wdyt?

**Reply by mgalbright** [reply]
That looks problematic to me -- it looks like it makes the forecasts inaccurate:  When I run your code with "+ response.UtcOffsetSeconds()" (with the America/Chicago time zone), the forecast shows the daily high temperature at my location occurring at daytime_local = "2024-02-12 08:00:00-06:00", i.e. 8 am in my local time zone, which is inaccurate.  In reality, the high temperature should occur about 2PM in the afternoon in my local time zone according to other weather forecast apps (after the sun shined all day, warming the air).

When I run my code (without the "+ response.UtcOffsetSeconds()"), the forecast shows the daily high temperature occurring daytime_local = "2024-02-12 14:00:00-06:00", i.e. 2PM local time as expected.
